crates/node/src/txvalidation/mod.rs

```diff
@@ -0,0 +1,199 @@
+use crate::txvalidation::event::{ReceivedTx, TxEvent};
+use crate::types::{
+    transaction::{Created, Received, Validated},
+    Transaction,
+};
+use crate::Mempool;
+use futures_util::Stream;
+use futures_util::TryFutureExt;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::marker::PhantomData;
+use std::net::SocketAddr;
+use std::path::PathBuf;
+use std::sync::Arc;
+use thiserror::Error;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::UnboundedSender;
+use tokio::sync::oneshot;
+use tokio::sync::RwLock;
+use tokio::task::JoinHandle;
+use tokio_stream::wrappers::UnboundedReceiverStream;
+
+pub mod acl;
+mod download_manager;
+mod event;
+
+#[allow(clippy::enum_variant_names)]
+#[derive(Error, Debug)]
+pub enum EventProcessError {
+    #[error("Fail to rcv Tx from the channel: {0}")]
+    RcvChannelError(#[from] tokio::sync::oneshot::error::RecvError),
+    #[error("Fail to send the Tx on the channel: {0}")]
+    SendChannelError(
+        #[from]
+        tokio::sync::mpsc::error::SendError<(Transaction<Received>, Option<CallbackSender>)>,
+    ),
+    #[error("Fail to send the Tx on the channel: {0}")]
+    PropagateTxError(#[from] Box<tokio::sync::mpsc::error::SendError<Transaction<Validated>>>),
+    #[error("validation fail: {0}")]
+    ValidateError(String),
+    #[error("Tx asset fail to download because {0}")]
+    DownloadAssetError(String),
+    #[error("Save Tx error: {0}")]
+    SaveTxError(String),
+    #[error("AclWhite list authenticate error: {0}")]
+    AclWhiteListAuthError(#[from] acl::AclWhiteListError),
+}
+
+pub type CallbackSender = oneshot::Sender<Result<(), EventProcessError>>;
+
+//Sending Tx interface.
+//Some marker type to define the sender source.
+pub struct RpcSender;
+#[derive(Clone)]
+pub struct P2pSender;
+pub struct TxResultSender;
+
+//use to send a tx to the event process.
+#[derive(Debug, Clone)]
+pub struct TxEventSender<T> {
+    sender: UnboundedSender<(Transaction<Received>, Option<CallbackSender>)>,
+    _marker: PhantomData<T>,
+}
+
+//Manage send from the p2p source
+impl TxEventSender<P2pSender> {
+    pub fn build(sender: UnboundedSender<(Transaction<Received>, Option<CallbackSender>)>) -> Self {
+        TxEventSender {
+            sender,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn send_tx(&self, tx: Transaction<Created>) -> Result<(), EventProcessError> {
+        self.sender
+            .send((tx.into_received(Received::P2P), None))
+            .map_err(|err| err.into())
```

Comment:
The naming in this module is a bit backwards I think. There's a lot of "send" in various type and method names, but they mostly deal with *received* data.

I think this naming should be improved, but let's keep the present form as-is in this PR and re-consider it afterwards.